pkg/provider/beyondtrust/provider.go

```diff
@@ -0,0 +1,327 @@
+func (*Provider) DeleteSecret(_ context.Context, _ esv1beta1.PushSecretRemoteRef) error {
+	return fmt.Errorf(errNotImplemented)
+}
+
+// GetSecretMap implements v1beta1.SecretsClient.
+func (*Provider) GetSecretMap(_ context.Context, _ esv1beta1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	return make(map[string][]byte), fmt.Errorf(errNotImplemented)
+}
+
+// PushSecret implements v1beta1.SecretsClient.
+func (*Provider) PushSecret(_ context.Context, _ *v1.Secret, _ esv1beta1.PushSecretData) error {
+	return fmt.Errorf(errNotImplemented)
+}
+
+// Validate implements v1beta1.SecretsClient.
+func (p *Provider) Validate() (esv1beta1.ValidationResult, error) {
+	timeout := 15 * time.Second
+	clientURL := p.apiURL
+
+	if err := esoClient.NetworkValidate(clientURL, timeout); err != nil {
+		ESOLogger.Error(err, "Network Validate", "clientURL:", clientURL)
+		return esv1beta1.ValidationResultError, err
+	}
+
+	return esv1beta1.ValidationResultReady, nil
+}
+
+func (*Provider) SecretExists(_ context.Context, _ esv1beta1.PushSecretRemoteRef) (bool, error) {
+	return false, fmt.Errorf(errNotImplemented)
+}
+
+// NewClient this is where we initialize the SecretClient and return it for the controller to use.
+func (p *Provider) NewClient(ctx context.Context, store esv1beta1.GenericStore, kube client.Client, namespace string) (esv1beta1.SecretsClient, error) {
+	config := store.GetSpec().Provider.Beyondtrust
+	logger := logging.NewLogrLogger(&ESOLogger)
+	apiURL := config.APIURL
+	certificate := ""
+	certificateKey := ""
+	clientTimeOutInSeconds := 45
+	retryMaxElapsedTimeMinutes := 15
+	separator := "/"
+
+	if config.Separator != "" {
+		separator = config.Separator
+	}
+
+	if config.Clienttimeoutseconds != 0 {
+		clientTimeOutInSeconds = config.Clienttimeoutseconds
+	}
+
+	backoffDefinition := backoff.NewExponentialBackOff()
+	backoffDefinition.InitialInterval = 1 * time.Second
+	backoffDefinition.MaxElapsedTime = time.Duration(retryMaxElapsedTimeMinutes) * time.Second
+	backoffDefinition.RandomizationFactor = 0.5
+
+	clientID, err := loadConfigSecret(ctx, config.Clientid, kube, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	clientSecret, err := loadConfigSecret(ctx, config.Clientsecret, kube, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	if config.Certificate != nil && config.Certificatekey != nil {
+		loadedCertificate, err := loadConfigSecret(ctx, config.Certificate, kube, namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		certificate = loadedCertificate
+
+		loadedCertificateKey, err := loadConfigSecret(ctx, config.Certificatekey, kube, namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		certificateKey = loadedCertificateKey
+	}
+
+	// validate inputs
+	errorsInInputs := utils.ValidateInputs(clientID, clientSecret, &apiURL, clientTimeOutInSeconds, &separator, config.VerifyCA, logger, certificate, certificateKey, &retryMaxElapsedTimeMinutes, &maxFileSecretSizeBytes)
```

Comment:
Please use a struct to group function parameters.

Reply:
Solved